client/src/components/Navbar.js

```diff
@@ -2,6 +2,79 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { Nav, Navbar, NavItem, NavDropdown, MenuItem } from 'react-bootstrap';
 import { Link } from 'react-router';
+import { connect } from 'react-redux';
+import { logSignerOut } from '../redux/actions/signature';
+
+const UserIsLoggedIn = props => {
```

Comment:
Most of UserIsLoggedIn and UserIsNotLogged in jsx is repeated in hard code.  This is bad for maintainability and also increase the fragility for future NavBar development.  a future developer would have to know to change both and make the same change to both if there ever be a change to the rest of the NavBar(for example if Why because a drop down or an optionally shown nav item).

I think a better way to do this is have just the login NavItem for login/log out be decided by your logic instead of the whole nav bar.